### PR TITLE
fix string format arguments

### PR DESCRIPTION
**Pull request by pinzart90**
https://jira.autodesk.com/browse/DYN-3856
String.Format exception thrown

### CODE REVIEW

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs [hunk 1]
```diff
@@ -498,18 +498,19 @@ internal void ExecutePackageDownload(string name, PackageVersion package, string
             var pmExt = DynamoViewModel.Model.GetPackageManagerExtension();
             if (result == MessageBoxResult.OK)
             {
+                package.full_dependency_ids.Count == package.full_dependency_versions.Count
```
Comment (pinzart90) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs:501]: I wonder what the best assert utility would be to check that these 2 arrays are in sync...
@aparajit-pratap any ideas ?

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs [hunk 2]
```diff
@@ -498,18 +498,19 @@ internal void ExecutePackageDownload(string name, PackageVersion package, string
...
                     try
                     {
-                        var depVersion = package.full_dependency_versions[i];
                         return Model.GetPackageVersionHeader(dep._id, depVersion);
                     }
                     catch
                     {
                         MessageBoxService.Show(
-                            String.Format(Resources.MessageFailedToDownloadPackageVersion, dep._id),
+                            String.Format(Resources.MessageFailedToDownloadPackageVersion, dep._id, depVersion),
```
Comment (pinzart90) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs:513]: THe message looks like this
```
Failed to download version {0} of package with id: {1}.  Please try again and report the package if you continue to have problems.
```

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs (hunk 1 above)
Comment (aparajit-pratap) [reply src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs:501]: `Debug.Assert`?

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs [hunk 3]
```diff
@@ -498,18 +498,19 @@ internal void ExecutePackageDownload(string name, PackageVersion package, string
...
                     try
                     {
-                        var depVersion = package.full_dependency_versions[i];
                         return Model.GetPackageVersionHeader(dep._id, depVersion);
                     }
                     catch
                     {
                         MessageBoxService.Show(
-                            String.Format(Resources.MessageFailedToDownloadPackageVersion, dep._id),
+                            String.Format(Resources.MessageFailedToDownloadPackageVersion, depVersion, dep._id),
```
Comment (pinzart90) [inline src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs:513]: ```Failed to download version {0} of package with id: {1}.  Please try again and report the package if you continue to have problems.``

**Review by aparajit-pratap — COMMENTED** [review]
How do these changes avert the crash?

**Comment by pinzart90** [comment]
Calling String.Format with the wrong number of arguments will cause an exception to be thrown.

Resources.MessageFailedToDownloadPackageVersion expects 2 arguments

File: src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs (hunk 3 above)
Comment (aparajit-pratap) [reply src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs:513]: Did it crash here at `dep._id` before? Was `dep` `null` in that case?

Comment (pinzart90) [reply src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerClientViewModel.cs:513]: Calling `String.Format` with the wrong number of arguments will cause an exception to be thrown. Resources.MessageFailedToDownloadPackageVersion expects 2 args and we were only passing one (dep._id)

**Review by aparajit-pratap — APPROVED** [review]

**Pull request merged**